homeassistant/components/deconz/__init__.py

```diff
@@ -186,3 +187,92 @@ def deconz_shutdown(event):
         entity_picture="/static/images/logo_deconz.jpeg",
         submit_caption="I have unlocked the gateway",
     )
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class DeconzFlowHandler(config_entries.ConfigFlowHandler):
+    """Handle a deCONZ config flow."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize the deCONZ flow."""
+        self.bridges = []
+        self.deconz_config = {}
+
+    async def async_step_init(self, user_input=None):
+        """Handle a flow start."""
+        from pydeconz.utils import async_discovery
+
+        if DOMAIN in self.hass.data:
+            return self.async_abort(
+                reason='one_instance_only'
+            )
+
+        if user_input is not None:
+            for bridge in self.bridges:
+                if bridge[CONF_HOST] == user_input[CONF_HOST]:
+                    self.deconz_config = bridge
+                    return await self.async_step_link()
+
+        session = aiohttp_client.async_get_clientsession(self.hass)
+        self.bridges = await async_discovery(session)
+
+        if len(self.bridges) == 1:
+            self.deconz_config = self.bridges[0]
+            return await self.async_step_link()
+        elif len(self.bridges) > 1:
+            hosts = []
+            for bridge in self.bridges:
+                hosts.append(bridge[CONF_HOST])
+            return self.async_show_form(
+                step_id='init',
+                data_schema=vol.Schema({
+                    vol.Required(CONF_HOST): vol.In(hosts)
+                })
+            )
+
+        return self.async_abort(
+            reason='no_bridges'
+        )
+
+    async def async_step_link(self, user_input=None):
+        """Attempt to link with the deCONZ bridge."""
+        from pydeconz.utils import async_get_api_key
+        errors = {}
+
+        if user_input is not None:
+            api_key = await async_get_api_key(
+                self.hass.loop, **self.deconz_config)
+            if api_key:
+                self.deconz_config[CONF_API_KEY] = api_key
+                return self.async_create_entry(
+                    title='deCONZ',
+                    data=self.deconz_config
+                )
+            else:
+                errors['base'] = 'no_key'
+
+        return self.async_show_form(
+            step_id='link',
+            errors=errors,
+        )
+
+
+async def async_setup_entry(hass, entry):
+    """Set up a bridge for a config entry."""
+    result = await async_setup_deconz(hass, None, entry.data)
+    if result:
+        return True
+    return False
+
+
+async def async_unload_entry(hass, entry):
```

Comment:
Remove this. We don't support unloading platforms yet and so you are not removing all the entities that this config entry has created.